### PR TITLE
Rewrote Nav Drawer to Kotlin + Dark Theme

### PR DESCRIPTION
**Pull request by NoodleMage**
## Light Theme
screener_20160317 200711
screener_20160317 092229
screener_20160314 220500

## Settings Option
screener_20160317 092238

## Dark Theme
screener_20160317 201154
screener_20160317 092334
screener_20160314 220452

### CODE REVIEW

**Comment by NoodleMage** [comment]
Tested for SDK 16 / 21 / Android N.

**Comment by inorichi** [comment]
"My library" is a bit unreadable in the picture with the black theme IMO.

**Comment by NoodleMage** [comment]
I agree. But this is because the primary color isn't material.  Thus the only way around this is using a different primary color for the dark theme.

**Comment by inorichi** [comment]
Okay. Any particular reason to remove `app/src/main/res/drawable-xhdpi/card_background.9.png`? It provides a grey shadow to the covers. It also looks fine in the dark theme.

**Comment by NoodleMage** [comment]
I thought the background is white in dark theme when I use card_background. Can't test it at this time

**Comment by inorichi** [comment]
Not very noticeable but it seems to look better:

**Comment by NoodleMage** [comment]
Does the ripple effect on touch still work? If so will change it back then ;)

**Comment by inorichi** [comment]
Yep, it works.

**Comment by inorichi** [comment]
Let me do some review/changes. Then just pull my changes to your branch if they are fine.

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/pager/PagerReaderFragment.kt [hunk 1]
```diff
@@ -74,12 +74,12 @@ class PagerReaderFragment : BaseFragment() {
     /**
      * Text color for black theme.
      */
-    private val lightGreyColor by lazy { ContextCompat.getColor(context, R.color.light_grey) }
+    private val lightGreyColor by lazy { ContextCompat.getColor(context, android.R.attr.textColorHint) }
```
Comment (inorichi) [inline app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/pager/PagerReaderFragment.kt]: This line makes the reader to crash when it's opened (only when using black background).

**Comment by inorichi** [comment]
The only issue I see now is the drawer items colors and the text in the dark theme. I think we should fix this before merging. Also pull my changes from `theme_update2` when you can.

**Comment by NoodleMage** [comment]
Pulled. Maybe we can use #607D8B as primary for the dark theme? What do you mean by text?

**Comment by inorichi** [comment]
Same as my first comment: "My library" is a bit unreadable in the picture with the black theme IMO.

File: app/src/main/res/values/attrs.xml [hunk 2]
```diff
@@ -20,4 +20,6 @@
 
     <attr name="selectable_list_drawable" format="reference|integer" />
     <attr name="divider_drawable" format="reference|integer" />
+    <attr name="status_bar_trans" format="reference|color"/>
+    colorPrimaryDarkTrans
```
Comment (inorichi) [inline app/src/main/res/values/attrs.xml]: Uh... and no warning? :)

Comment (NoodleMage) [inline app/src/main/res/values/attrs.xml]: Nope. Themes.xml just rolls with it xD

**Pull request merged**